BepInEx.Core/Bootstrap/BaseChainloader.cs

```diff
@@ -109,23 +116,66 @@
         return pluginTarget.Build > CurrentAssemblyVersion.Build;
     }
 
+    /// <summary>
+    ///     Obtains the <see cref="PluginInfo"/> corresponding to its GUID
+    /// </summary>
+    /// <param name="pluginGuid">The GUID of the plugin to get the info from</param>
+    /// <param name="foundPluginInfo">When this method returns, contains the <see cref="PluginInfo"/>
+    /// associated with the specified GUID, if the plugin was loaded; otherwise, null. This parameter is passed uninitialized.</param>
+    /// <returns>true if the plugin with the GUID was loaded by the chainloader; otherwise, false.</returns>
+    public bool TryGetPluginInfoFromGuid(string pluginGuid, out PluginInfo foundPluginInfo)
```

Comment:
Because after discussion with manly, it was decided that in an api cleanup pr, `plugins` won't be public anymore, but rather, a read only collection of the plugins will be. This method could still be used to lookup from GUID.

Reply:
Isn't `IReadOnlyCollection<PluginInfo>` + `TryGetPluginInfoFromGuid` literally the same api surface as `IReadOnlyDictionary<string, PluginInfo>`?

Reply:
After discussions, I decided for this pr to use the TryGet directly, but we will have `Plugins`'s type change to be a readonly version of the dictionnary through some polyfills (due to ReadOnlyDictionnary not being avilable in net35)